weblate/api/tests.py

```diff
@@ -2391,6 +2391,29 @@ def test_create_component_enforced(self) -> None:
         component = Component.objects.get(slug="local-project")
         self.assertEqual(component.enforced_checks, ["same"])
 
+    def test_create_component_with_file_format_params(self) -> None:
+        self.do_request(
+            "api:project-components",
+            self.project_kwargs,
+            method="post",
+            code=201,
+            superuser=True,
+            request={
+                "name": "API project",
+                "slug": "api-project",
+                "repo": self.format_local_path(self.git_repo_path),
+                "filemask": "po/*.po",
+                "file_format": "po",
+                "file_format_params": {"po_line_wrap": -1, "yaml_indent": 8},
+                "push": "https://username:password@github.com/example/push.git",
+                "new_lang": "none",
+            },
+            format="json",
+        )
+        component = Component.objects.get(slug="api-project", project__slug="test")
+        self.assertEqual(component.file_format_params["po_line_wrap"], -1)
+        self.assertNotIn("yaml_indent", component.file_format_params)
```

Comment:
Can you please add a test case for using a parameter with an invalid value as well?

Reply:
Also, I'm not sure if swallowing unknown parameters is a good idea here. It can cause silent breakages, failing with ValidationError in this case might be more desired.